src/components/config-tabs/tab-contents/DenominatorRelations.js

```diff
@@ -10,13 +11,158 @@ import {
     const [relations, setRelations] = useState(null)
+    const [isModalHidden, setIsModalHidden] = useState(true)
+    const [newDenominatorRelationInfo, setNewDenominatorRelationInfo] =
+        useState({
+            A: '',
+            B: '',
+            code: '',
+            criteria: 12,
+            name: '',
+            type: '',
+        })
+    const [selectedDenominatorA, setSelectedDenominatorA] = useState('')
+    const [selectedDenominatorB, setSelectedDenominatorB] = useState('')
+    const [filteredDenominators, setFilteredDenominators] = useState([])
+    const [isLoading, setIsLoading] = useState(false)
+    const [updateType, setUpdateType] = useState('')
+
+    const handleDenominatorTypeChange = (type) => {
+        // reset the selected denominators
+        setSelectedDenominatorA('')
+        setSelectedDenominatorB('')
+
+        // filter denominators by type
+        setFilteredDenominators(
+            filterDenominatorsByType(configurations.denominators, type)
+        )
+        setNewDenominatorRelationInfo({
+            ...newDenominatorRelationInfo,
+            type: type,
+        })
+    }
+
+    const onSaveDenominatorUpdates = async (updateType) => {
+        //show a loader while updating
+        setIsLoading(true)
+
+        if (updateType === 'create') {
+            // generate and set the new denominator code for create only
+            const newDenominatorCode = generateDenominatorRelationCode(
+                configurations.denominatorRelations
+            )
+
+            const updatedConfigurations = updateDenominatorRelations({
+                configurations,
+                relation: newDenominatorRelationInfo,
+                updateType,
+                newCode:newDenominatorCode,
+            })
+            const response = await mutate({
+                configurations: updatedConfigurations,
+            })
+
+            console.log(updatedConfigurations)
+            if (response) {
+                //stop the loader after updating
+                setRelations([...relations, newDenominatorRelationInfo])
+                setIsLoading(false)
+            }
+            setIsModalHidden(true)
+        } else if (updateType === 'update') {
+            const updatedConfigurations = updateDenominatorRelations({
+                configurations,
+                relation:newDenominatorRelationInfo,
+                updateType,
+                newCode: null,
+            })
+            const response = await mutate({
+                configurations: updatedConfigurations,
+            })
+            if (response) {
+                //stop the loader after updating
+                setIsLoading(false)
+            }
+            setIsModalHidden(true)
+        }
+    }
+
+    const onEditRelation = (relation) => {
+        setIsModalHidden(false)
+        setNewDenominatorRelationInfo(relation)
+        setSelectedDenominatorA(relation.A)
```

Comment:
I think that you might need to get the filtered relations first and then only set A and B if A and B are in those filtered relations...
We have some situations (I believe because the old app did not filter based on `un` type, where there can be a mismatch between the A/B denominators and the relation type; I guess this could also occur if you change the type of a denominator after you create a denominator relation).
Anyway, this potential for mismatch will currently cause an error (e.g. try clicking on the first denominator relation currently in the list (Population < 1 - official vs UN) and you get an error: `Error: There is no option with the value: "P1". Make sure that the value passed to the selected prop matches the value of an existing option.`)